algosdk/atomic_transaction_composer.py

```diff
@@ -0,0 +1,529 @@
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            stxn = txn_group[i].sign(self.private_key)
+            stxns.append(stxn)
+        return stxns
+
+
+class LogicSigTransactionSigner(TransactionSigner):
+    """
+    Represents a Transaction Signer for a LogicSig that can sign transactions from an
+    atomic transaction group.
+
+    Args:
+        lsig (LogicSigAccount): LogicSig account
+    """
+
+    def __init__(self, lsig) -> None:
+        super().__init__()
+        self.lsig = lsig
+
+    def sign_transactions(self, txn_group, indexes):
+        """
+        Sign transactions in a transaction group given the indexes.
+
+        Returns an array of encoded signed transactions. The length of the
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            stxn = transaction.LogicSigTransaction(txn_group[i], self.lsig)
+            stxns.append(stxn)
+        return stxns
+
+
+class MultisigTransactionSigner(TransactionSigner):
+    """
+    Represents a Transaction Signer for a Multisig that can sign transactions from an
+    atomic transaction group.
+
+    Args:
+        msig (Multisig): Multisig account
+        sks (str): private keys of multisig
+    """
+
+    def __init__(self, msig, sks) -> None:
+        super().__init__()
+        self.msig = msig
+        self.sks = sks
+
+    def sign_transactions(self, txn_group, indexes):
+        """
+        Sign transactions in a transaction group given the indexes.
+
+        Returns an array of encoded signed transactions. The length of the
+        array will be the same as the length of indexesToSign, and each index i in the array
+        corresponds to the signed transaction from txnGroup[indexesToSign[i]].
+
+        Args:
+            txn_group (list[Transaction]): atomic group of transactions
+            indexes (list[int]): array of indexes in the atomic transaction group that should be signed
+        """
+        stxns = []
+        for i in indexes:
+            mtxn = transaction.MultisigTransaction(txn_group[i], self.msig)
+            for sk in self.sks:
+                mtxn.sign(sk)
+            stxns.append(mtxn)
+        return stxns
+
+
```

Comment:
These method-less classes are good candidates for becoming [data classes](https://realpython.com/python-data-classes/). Just a thought.

Reply:
I believe we have to (unfortunately) support Python 3.5 in this SDK: https://github.com/algorand/py-algorand-sdk/blob/8cd3bb615ca3c1c15846bd9eb7ec180ac4267fc4/setup.py#L21

This is also why we couldn't use f-strings (just one version short).

Reply:
ahhh dang it! 

So no `f-strings` either (only [since 3.6](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python) )